### ISSUE TITLE
training using my own dataset，after sever epoch loss become higher and unchange

### ISSUE THREAD

**Issue by lengchenyu** [zylo117/Yet-Another-EfficientDet-Pytorch#78]
![image](https://user-images.githubusercontent.com/13712994/79288244-ce847d80-7ef8-11ea-97e9-7179150ddb75.png)

![image](https://user-images.githubusercontent.com/13712994/79288262-d5ab8b80-7ef8-11ea-945f-2b4863b9cd2f.png)

as the picture show， after 6-7 epoch， the loss become higher and totoal  loss still as 2.30212 always。

**Comment by prfans** [comment]
Maybe, You can try to reduce lr, and examinate data structrue?

**Comment by zylo117** [comment]
it could be dataset corruption or mislabel, you should visualize your gt boxes and their labels

**Comment by lengchenyu** [comment]
> it could be dataset corruption or mislabel, you should visualize your gt boxes and their labels

for two reason i think is not the dataset problem
1） because in early epoch the loss is normal，and for each epoch will iterate all the data，if my dataset have problem，it will show abnormal in early training。

2）i use the same dataset train in another project ：https://github.com/signatrix/efficientdet 。And the loss is normal，no such problem。

**Comment by zylo117** [comment]
> > it could be dataset corruption or mislabel, you should visualize your gt boxes and their labels
> 
> for two reason i think is not the dataset problem
> 1） because in early epoch the loss is normal，and for each epoch will iterate all the data，if my dataset have problem，it will show abnormal in early training。
> 
> 2）i use the same dataset train in another project ：https://github.com/signatrix/efficientdet 。And the loss is normal，no such problem。And i test the trained model in my test data，it show normal。
> 
> And the core problem is the cls loss unchange after sever epoch。

Oh, I guess I know what is going on.

It's exploding gradients. The lr is too high at first, and then it explodes and it can hardly be undone.

You should try a smaller lr and train again before the explosion.

**Comment by lengchenyu** [comment]
> > > it could be dataset corruption or mislabel, you should visualize your gt boxes and their labels
> > 
> > 
> > for two reason i think is not the dataset problem
> > 1） because in early epoch the loss is normal，and for each epoch will iterate all the data，if my dataset have problem，it will show abnormal in early training。
> > 2）i use the same dataset train in another project ：https://github.com/signatrix/efficientdet 。And the loss is normal，no such problem。And i test the trained model in my test data，it show normal。
> > And the core problem is the cls loss unchange after sever epoch。
> 
> Oh, I guess I know what is going on.
> 
> It's exploding gradients. The lr is too high at first, and then it explodes and it can hardly be undone.
> 
> You should try a smaller lr and train again before the explosion.

ok, thanks. I will  try a smaller lr train again
